examples/model_selection/plot_precision_recall.py

```diff
@@ -71,76 +92,100 @@
+# --------------------------------------------------------
+#
+# We create a multiclass data, to illustrate the precision-recall in
+# multiclass settings
 
+import numpy as np
 from sklearn import svm, datasets
-from sklearn.metrics import precision_recall_curve
-from sklearn.metrics import average_precision_score
 from sklearn.model_selection import train_test_split
 from sklearn.preprocessing import label_binarize
+# We use OneVsRestClassifier for multi-label prediction
 from sklearn.multiclass import OneVsRestClassifier
 
 # import some data to play with
 iris = datasets.load_iris()
 X = iris.data
 y = iris.target
 
-# setup plot details
-colors = cycle(['navy', 'turquoise', 'darkorange', 'cornflowerblue', 'teal'])
-lw = 2
-
-# Binarize the output
-y = label_binarize(y, classes=[0, 1, 2])
-n_classes = y.shape[1]
+# Binarize the output (multi-label like settings)
+Y = label_binarize(y, classes=[0, 1, 2])
+n_classes = Y.shape[1]
 
 # Add noisy features
 random_state = np.random.RandomState(0)
 n_samples, n_features = X.shape
 X = np.c_[X, random_state.randn(n_samples, 200 * n_features)]
 
 # Split into training and test
-X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=.5,
+X_train, X_test, Y_train, Y_test = train_test_split(X, Y, test_size=.5,
                                                     random_state=random_state)
 
 # Run classifier
 classifier = OneVsRestClassifier(svm.SVC(kernel='linear', probability=True,
-                                 random_state=random_state))
-y_score = classifier.fit(X_train, y_train).decision_function(X_test)
+                     random_state=random_state))
+classifier.fit(X_train, Y_train)
+y_score = classifier.decision_function(X_test)
+
+
+###############################################################################
+# Compute Precision-Recall score
+# -------------------------------
+from sklearn.metrics import precision_recall_curve
+from sklearn.metrics import average_precision_score
 
-# Compute Precision-Recall and plot curve
+# For each class
 precision = dict()
 recall = dict()
 average_precision = dict()
 for i in range(n_classes):
-    precision[i], recall[i], _ = precision_recall_curve(y_test[:, i],
+    precision[i], recall[i], _ = precision_recall_curve(Y_test[:, i],
                                                         y_score[:, i])
-    average_precision[i] = average_precision_score(y_test[:, i], y_score[:, i])
+    average_precision[i] = average_precision_score(Y_test[:, i], y_score[:, i])
 
-# Compute micro-average ROC curve and ROC area
-precision["micro"], recall["micro"], _ = precision_recall_curve(y_test.ravel(),
+# A "micro-average": quantifying score on all classes jointly
+precision["micro"], recall["micro"], _ = precision_recall_curve(Y_test.ravel(),
     y_score.ravel())
-average_precision["micro"] = average_precision_score(y_test, y_score,
+average_precision["micro"] = average_precision_score(Y_test, y_score,
```

Comment:
Is this standard? I'm not sure I understand what that means. Maybe do binary for this example?